ironhawk/main.go

```diff
@@ -76,14 +82,48 @@ func Run(host string, port int) {
 
 		if strings.HasSuffix(strings.ToUpper(args[0]), ".WATCH") {
 			fmt.Println("entered the watch mode for", c.Cmd, strings.Join(c.Args, " "))
+			
+			// Create a watch-specific signal channel
+			watchSigChan := make(chan os.Signal, 1)
+			signal.Notify(watchSigChan, os.Interrupt)
+			
 			ch, err := client.WatchCh()
 			if err != nil {
 				fmt.Println("error watching:", err)
 				continue
 			}
-			for resp := range ch {
-				renderResponse(resp)
+
+			// Set watch mode flag
+			inWatchMode = true
+			
+			// Create done channel for watch loop
+			done := make(chan struct{})
+			
+			// Start watch loop in goroutine
+			go func() {
+				for resp := range ch {
+					select {
+					case <-done:
```

Comment:
Once we exit the WATCH mode, we also need to [UNWATCH](https://dicedb.io/commands/unwatch/) the current subscription. Also, can you restructure and rearrange the code for better readability?